sdk/identity/Azure.Identity/src/AzureCliCredentialClient.cs

```diff
@@ -0,0 +1,172 @@
+            string argument = string.Empty;
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                fileName = "cmd";
+                argument = $"/c \"{command}\"";
+            }
+            else
+            {
+                fileName = "/bin/sh";
+                argument = $"-c \"{command}\"";
+            }
+
+            ProcessStartInfo procStartInfo = new ProcessStartInfo()
+            {
+                FileName = fileName,
+                Arguments = argument,
+                UseShellExecute = false,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                CreateNoWindow = true,
+                WorkingDirectory = _workingDir,
+                Environment = { { "PATH", _path } }
+            };
+
+            return await RunProcessAsync(async, procStartInfo, cancellationToken).ConfigureAwait(false);
+        }
+
+        private static async ValueTask<(string, int)> RunProcessAsync(bool async, ProcessStartInfo procStartInfo, CancellationToken cancellationToken)
+        {
+            int exitCode = default;
+            StringBuilder stdOutput = new StringBuilder();
+            StringBuilder stdError = new StringBuilder();
+            bool completed;
+
+            using (Process proc = new Process())
+            {
+
+                proc.StartInfo = procStartInfo;
+                proc.OutputDataReceived += new DataReceivedEventHandler((sender, e) => stdOutput.AppendLine(e.Data));
+                proc.ErrorDataReceived += new DataReceivedEventHandler((sender, e) => stdError.AppendLine(e.Data));
+
+                proc.Start();
+                proc.BeginOutputReadLine();
+                proc.BeginErrorReadLine();
+
+                completed = async ? await Task.Run(() => proc.WaitForExit(CliProcessTImeoutMs), cancellationToken).ConfigureAwait(false) : proc.WaitForExit(CliProcessTImeoutMs);
+
+                if (!completed)
+                {
+                    proc.CancelOutputRead();
+
+                    proc.CancelErrorRead();
+
+                    throw new AuthenticationFailedException(AzureCliTimeoutError);
+                }
+
+                exitCode = proc.ExitCode;
+            }
+
+            if (exitCode != 0)
+            {
+                return (stdError.ToString(), exitCode);
+            }
+            else
+            {
+                return (stdOutput.ToString(), exitCode);
+            }
+        }
+
+        private static (string, string) GetSafePathAndWorkingDirectory()
+        {
+            string path = EnvironmentVariables.Path;
+
+            if (string.IsNullOrEmpty(path))
+            {
+                path = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? DefaultPathWindows : DefaultPath;
+            }
+
+            string workingDir = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? path.Split(';')?[0] : path.Split(':')?[0];
```

Comment:
You should pass the param to `Split` to collapse subsequent delimiters so you don't accidentally grab an empty path (not sure what would happen, but can't imagine it would be good). But why the null coalescer? Neither of these should result in null or the workingDir would likely define to the CWD, which we don't want. I'd be more explicit here.